code/modules/projectiles/gun.dm

```diff
@@ -21,9 +21,9 @@
 	attack_verb = list("struck", "hit", "bashed")
 
 	var/fire_sound = "gunshot"
-	var/suppressed = 0					//whether or not a message is displayed when fired
-	var/can_suppress = 0
-	var/can_unsuppress = 1
+	var/suppressed = FALSE					//whether or not a message is displayed when fired
```

Comment:
This should be null instead if it holds the suppressor reference. (Also maybe renaming it to just supressor would be good idea)

Reply:
Renaming it would require renaming all 40+ suppressed state icons in their dmi files, and for anyone downstream to duplicate it for any custom weapons too. Don't think it's worth doing really.